remote/remote_state.go

```diff
@@ -52,57 +53,34 @@ func (remoteState *RemoteState) Initialize(terragruntOptions *options.Terragrunt
 	return nil
 }
 
-// Configure Terraform remote state
-func (remoteState RemoteState) ConfigureRemoteState(terragruntOptions *options.TerragruntOptions) error {
-	shouldConfigure, err := shouldConfigureRemoteState(remoteState, terragruntOptions)
-	if err != nil {
-		return err
-	}
-
-	if shouldConfigure {
-		terragruntOptions.Logger.Printf("Initializing remote state for the %s backend", remoteState.Backend)
-		if err := remoteState.Initialize(terragruntOptions); err != nil {
-			return err
-		}
-
-		terragruntOptions.Logger.Printf("Configuring remote state for the %s backend", remoteState.Backend)
-		return shell.RunTerraformCommand(terragruntOptions, initCommand(remoteState)...)
-	}
-
-	return nil
-}
-
 // Returns true if remote state needs to be configured. This will be the case when:
 //
 // 1. Remote state has not already been configured
 // 2. Remote state has been configured, but for a different backend type, and the user confirms it's OK to overwrite it.
-func shouldConfigureRemoteState(remoteStateFromTerragruntConfig RemoteState, terragruntOptions *options.TerragruntOptions) (bool, error) {
+func (remoteState *RemoteState) NeedsInit(terragruntOptions *options.TerragruntOptions) (bool, error) {
 	state, err := ParseTerraformStateFileFromLocation(terragruntOptions.WorkingDir)
 	if err != nil {
 		return false, err
 	}
 
 	if state != nil && state.IsRemote() {
-		return shouldOverrideExistingRemoteState(state.Backend, remoteStateFromTerragruntConfig, terragruntOptions)
-	} else {
-		return true, nil
+		return remoteState.differsFrom(state.Backend, terragruntOptions), nil
 	}
+	return true, nil
 }
 
-// Check if the remote state that is already configured matches the one specified in the Terragrunt config. If it does,
-// return false to indicate remote state does not need to be configured again. If it doesn't, prompt the user whether
-// we should override the existing remote state setting.
-func shouldOverrideExistingRemoteState(existingBackend *TerraformBackend, remoteStateFromTerragruntConfig RemoteState, terragruntOptions *options.TerragruntOptions) (bool, error) {
-	if existingBackend.Type != remoteStateFromTerragruntConfig.Backend {
-		prompt := fmt.Sprintf("WARNING: Terraform remote state is already configured, but for backend %s, whereas your Terragrunt configuration specifies %s. Overwrite?", existingBackend.Type, remoteStateFromTerragruntConfig.Backend)
-		return shell.PromptUserForYesNo(prompt, terragruntOptions)
+// Returns true if this remote state is different than the given remote state that is currently being used by terraform.
+func (remoteState *RemoteState) differsFrom(existingBackend *TerraformBackend, terragruntOptions *options.TerragruntOptions) bool {
+	if existingBackend.Type != remoteState.Backend {
+		terragruntOptions.Logger.Printf("Backend type has changed from %s to %s", existingBackend.Type, remoteState.Backend)
+		return true
```

Comment:
Hm, alright, if Terraform prompts about it, then that's probably good enough.

Renaming "Initialize" to "Provision" probably makes sense, but perhaps as part of a different PR, to prevent this one from getting much larger.